new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java

```diff
@@ -5,86 +5,175 @@
 
 import java.util.HashMap;
 
-import oracle.weblogic.kubernetes.extensions.LoggedTest;
-
-public class Helm implements LoggedTest {
-
-  private String chartName;
-  private String releaseName;
-  private String namespace;
-  private String repoUrl;
-  private HashMap<String, Object> values;
-
-  public Helm chartName(String chartName) {
-    this.chartName = chartName;
-    return this;
-  }
-
-  public Helm releaseName(String releaseName) {
-    this.releaseName = releaseName;
-    return this;
-  }
-
-  public Helm namespace(String namespace) {
-    this.namespace = namespace;
-    return this;
-  }
-
-  public Helm repoUrl(String repoUrl) {
-    this.repoUrl = repoUrl;
-    return this;
-  }
+import oracle.weblogic.kubernetes.utils.ExecCommand;
+import oracle.weblogic.kubernetes.utils.ExecResult;
+
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class Helm {
+
+  /**
+   * install helm chart
+   * @param params the parameters to helm as values
+   * @return true on success, false otherwise
+   */
+  public static boolean install(HelmParams params) {
+    String namespace = params.getNamespace();
+
+    // assertions for required parameters
+    assertThat(namespace)
+        .as("make sure namespace is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getReleaseName())
+        .as("make sure releaseName is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getRepoUrl() != null && params.getChartName() == null)
```

Comment:
These combinations are easier to read and debug if they are separate. Like:
assertNotNull(params.getRepoUrl(), "repo URL was not provided") 
or
assertThat(params.getRepoUrl())
.as("repo URL was not provided")
. isNotNull()
.isNotEmpty();

Reply:
@ddsharpe here I need to check chartName is also provided only when repoUrl is provided. Otherwise chartName is not required. 

When repoUrl is provided, here its adding a chart repository, for that both repoUrl  and chartName are required. hence the combination.

Reply:
checking for more readable ways ...

Reply:
@markxnelson @rjeberhard any suggestions please ^^ ?